File: lib/adminterface/extensions/inputs/color_input.rb
module Adminterface
  module Extensions
    module Inputs
      module ColorInput
        include Base
        include Base::Stringish
        prepend Base::Groupish

        def input_html
          builder.color_field(method, input_html_options)
        end

        def input_html_options
          opts = super
          opts[:class] = "#{opts[:class]} form-control-color".squish
          opts
        end
      end
    end
  end
end

module ActiveAdmin
  module Inputs
    class ColorInput < ::Formtastic::Inputs::ColorInput
      include ::Adminterface::Extensions::Inputs::ColorInput
    end
  end
end

File: app/javascript/adminterface/lib/inputs/color_input.js
import BaseInput from './base_input'

class ColorInput extends BaseInput {
  _defaultInputHTMLOptions () {
    const options = {
      class: `form-control form-control-color ${this.options.as}`,
      type: 'color'
    }

    return { ...super._defaultInputHTMLOptions(), ...options }
  }
}

export default ColorInput
